Firestore/core/src/firebase/firestore/api/firestore.mm

```diff
@@ -143,9 +144,23 @@
   // The client must be initialized to ensure that all subsequent API usage
   // throws an exception.
   EnsureClientConfigured();
+  client_running_ = false;
```

Comment:
This is marking the client no longer running before shutdown has completed. You need to do this only once `shutdownWithCallback` has completed. Specifically, this means that you can't just pass through this callback. You'll need to create a new lambda here that clears `client_running_` and then calls the supplied callback.

Also the thread safety concerns I've noted elsewhere apply to all usages of `client_running_`. Ensure that you're holding the `mutex_` when accessing it.